perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2999,7 +2999,11 @@ public List<ApplicationFormItemWithPrefilledValue> getFormItemsWithPrefilledValu
 
 		itemsWithValues.stream()
 			.filter(item -> item.getFormItem().getType() == EMBEDDED_GROUP_APPLICATION)
-			.forEach(item -> setGroupsToCheckBoxForGroups(sess, item, vo, group));
+			.forEach(item -> setGroupsToCheckBoxForGroups(sess, item, user, vo, group));
+		itemsWithValues = itemsWithValues.stream()
+			.filter(item -> item .getFormItem().getType() != EMBEDDED_GROUP_APPLICATION
+				|| item.getFormItem().getI18n().get(ApplicationFormItem.EN).getOptions() != null)
```

Comment:
Shouldn't be there the AND (`&&`)? I thought you wanted to collect groups that are not in form item with type EMBEDDED_GROUP_APPLICATION **AND** also have options in English, am I right?
```suggestion
			.filter(item -> item .getFormItem().getType() != EMBEDDED_GROUP_APPLICATION
				&& item.getFormItem().getI18n().get(ApplicationFormItem.EN).getOptions() != null)
```